mt940/models.py

```diff
@@ -381,7 +381,7 @@ def parse(self, data):
                     self.transactions.append(transaction)
                 else:
                     transaction.data.update(result)
-            elif tag.scope is Transaction and self.transactions:
```

Comment:
The first part of your suggestion is in line with my implementation. For the second part, please note that `class Sublass(Transaction)` `is` _not_ `class Transaction`. `issubclass` is needed to consider class inheritance.